stl/inc/ranges

```diff
@@ -8780,20 +8780,57 @@ namespace ranges {
         _EXPORT_STD inline constexpr _Zip_transform_fn zip_transform{};
     } // namespace views
 
+    template <class _Ty, size_t>
+    using _Repeat_type = _Ty;
+
     template <class _Ty, class _Indices>
     struct _Repeated_tuple_impl;
 
     template <class _Ty, size_t... _Indices>
     struct _Repeated_tuple_impl<_Ty, index_sequence<_Indices...>> {
-        template <size_t>
-        using _Repeat_type = _Ty;
-
-        using type = tuple<_Repeat_type<_Indices>...>;
+        using type = tuple<_Repeat_type<_Ty, _Indices>...>;
     };
 
     template <class _Ty, size_t _Nx>
     using _Repeated_tuple = typename _Repeated_tuple_impl<_Ty, make_index_sequence<_Nx>>::type;
 
+    template <class _Fn, class _Ty, class _Indices>
+    inline constexpr bool _Regular_invocable_with_repeated_type_impl = false;
+
+    template <class _Fn, class _Ty, size_t... _Indices>
+    inline constexpr bool _Regular_invocable_with_repeated_type_impl<_Fn, _Ty, index_sequence<_Indices...>> =
+        regular_invocable<_Fn, _Repeat_type<_Ty, _Indices>...>;
+
+    template <class _Fn, class _Ty, size_t _Nx>
+    concept _Regular_invocable_with_repeated_type =
+        _Regular_invocable_with_repeated_type_impl<_Fn, _Ty, make_index_sequence<_Nx>>;
+
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, VSO-1761088
+    template <class _Fn, class _Ty, class _Indices>
+    struct _Invoke_result_with_repeated_type_impl;
+
+    template <class _Fn, class _Ty, size_t... _Indices>
+    struct _Invoke_result_with_repeated_type_impl<_Fn, _Ty, index_sequence<_Indices...>> {
+        using type = invoke_result_t<_Fn, _Repeat_type<_Ty, _Indices>...>;

```

Comment:
The following workaround seemly works for MSVC ([Godbolt link](https://godbolt.org/z/nYvq9nhb8)):
```C++
using type = decltype(_STD invoke(_STD declval<_Fn>(), _STD declval<_Repeat_type<_Ty, _Indices>>()...));
```

Should we use this to avoid recursive template? (Perhaps this could also unblock some test cases for x86.)

Reply:
Done: https://github.com/microsoft/STL/pull/3546/commits/2d8174449879376d229a1263ad896d9df298dfc9.